crates/biome_service/src/configuration.rs

```diff
@@ -258,6 +259,38 @@ fn load_config(
     }
 }
 
+pub fn load_editorconfig(
+    file_system: &DynRef<'_, dyn FileSystem>,
+    workspace_root: PathBuf,
+) -> Result<
+    (
+        Option<PartialConfiguration>,
+        Vec<EditorConfigValidationError>,
+    ),
+    WorkspaceError,
+> {
+    if let Some(auto_search_result) =
+        match file_system.auto_search(&workspace_root, [".editorconfig"].as_slice(), false) {
+            Ok(Some(auto_search_result)) => Some(auto_search_result),
+            Ok(None) => None,
+            Err(error) => return Err(WorkspaceError::from(error)),
+        }
+    {
+        let AutoSearchResult {
+            content,
+            file_path: _,
+        } = auto_search_result;
+
+        let edconfig = biome_configuration::editorconfig::parse_str(&content)
+            .map_err(|err| DeserializationDiagnostic::new(format!("{}", err)))
+            .map_err(ConfigurationDiagnostic::Deserialization)?;
```

Comment:
I haven't had the chance to review the previous PR, but external errors that need to be mapped to our diagnostics must have an adapter.

An adapter is just a small typed wrapper for errors emitted by third party crates. Here's few examples:

The error emitted by `oxc_resolver`:
https://github.com/biomejs/biome/blob/32e422d0d63ec17c7eded814cdde0d26b158c110/crates/biome_diagnostics/src/adapters.rs#L111-L133

And this when it's called:
https://github.com/biomejs/biome/blob/32e422d0d63ec17c7eded814cdde0d26b158c110/crates/biome_configuration/src/diagnostics.rs#L88

The error emitted by `bpaf`:

https://github.com/biomejs/biome/blob/32e422d0d63ec17c7eded814cdde0d26b158c110/crates/biome_diagnostics/src/adapters.rs#L74-L108

And this is when it's called: 
https://github.com/biomejs/biome/blob/32e422d0d63ec17c7eded814cdde0d26b158c110/crates/biome_cli/src/diagnostics.rs#L317


These adapters must be the `source` of **our** diagnostic, in this case `EditorConfigValidationError` or a **new** diagnostic ad-hoc for editor config (it can be an enum). I wouldn't use `DeserializationDiagnostic` because that's specifically used for when we use `biome_deserialize`, which we aren't using (other than an helper, which is `StringSet`)

Reply:
Very useful, thank you.